tests/lib/devicetree/emul/src/emulator.c

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2022 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/emul.h>
+
+static int emul_init(const struct emul *emul, const struct device *parent)
```

Comment:
This is the hardest coding guideline to understand so just helping out, you can't use `emul` as both the type and the name of the variable. Suggest: `dev_emul` for the variable name. This comes from C++ where the `struct` key word is optional when declaring a variable, so you could write `emul emul` which is confusing.